src/ostree/ot-builtin-refs.c

```diff
@@ -60,7 +70,60 @@ static gboolean do_ref (OstreeRepo *repo, const char *refspec_prefix, GCancellab
           g_print ("%s\n", ref);
         }
     }
+  else if(opt_create)
+    {
+      gboolean old_ref_exists = FALSE;
+      const char *checksum = NULL;
+
+      g_hash_table_iter_init (&hashiter, refs);
+      while (g_hash_table_iter_next (&hashiter, &hashkey, &hashvalue))
+        {
+          g_autofree char *ref = NULL;
+          g_autofree char *remote = NULL;
+          const char *existing_ref= hashkey;
+
+          if (!ostree_parse_refspec (refspec_prefix, &remote, &ref, error))
+            goto out;
+
+          if(strcmp (opt_create, existing_ref) == 0)
+            {
+              g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
+                         "--create specified but ref %s already exists", existing_ref);
+              goto out;
+            }
+
+          if(strcmp(refspec_prefix, hashkey) == 0)
+            {
+              old_ref_exists = TRUE;
+              checksum = hashvalue;
+            }
+        }
+
+      if(old_ref_exists)
+        {
+          if (!ostree_repo_prepare_transaction (repo, NULL, cancellable, error))
```

Comment:
I'm not opposed to this but in practice I think it's saner to avoid the transactional ref bit - let's fix the bug that tempfiles stay around onerror first.